wagtail/admin/templatetags/tabs.py

```diff
@@ -0,0 +1,79 @@
+from django import template
+
+register = template.Library()
+
+
+def generate_tab_panel_id(title):
+    return f'tab-{title.lower().replace(" ",  "-")}'
+
+
+def generate_tab_label_id(title):
+    return f'tab-label-{title.lower().replace(" ",  "-")}'
+
+
+@register.tag
+def tabs(parser, token):
+    # Parse tabs and content
+    tabs = parser.parse(("endtabs",))
+    parser.delete_first_token()
+    return TabsNode(tabs, token)
+
+
+class TabsNode(template.Node):
+    def __init__(self, tabs, token):
+        self.tabs = tabs
+        self.token = token
+
+    def render(self, context):
+        output = f"""
+          <div class="tabs" data-controller="w-tabs" data-action="popstate@window->w-tabs#loadTabsFromHistory" data-w-tabs-active-tab-id-value=""> 
```

Comment:
We should be using Django template files for all generated html, not inline html like this.